src/test/java/org/kohsuke/github/GHWorkflowTest.java

```diff
@@ -77,9 +78,20 @@ public void testDisableEnable() throws IOException {
     }
 
     @Test
-    public void testDispatch() throws IOException {
+    public void testDispatch() throws IOException, IllegalArgumentException {
         GHWorkflow workflow = repo.getWorkflow("test-workflow.yml");
 
+        String expectedMessage = "ref cannot be null or empty";
+        IllegalArgumentException nullException = assertThrows(IllegalArgumentException.class, () -> {
+            workflow.dispatch(null);
+        });
+        assertThat(nullException.getMessage(), equalTo(expectedMessage));
+
+        IllegalArgumentException emptyException = assertThrows(IllegalArgumentException.class, () -> {
+            workflow.dispatch("");
+        });
+        assertThat(emptyException.getMessage(), equalTo(expectedMessage));
+
```

Comment:
```java
        // What happens in this case? 
        Exception emptyException = assertThrows(Exception.class, () -> {
            workflow.dispatch("invalidRef");
        });
```

Reply:
@bitwiseman I'm not super sure what would happen in this case... As of now, 'ref' as a string, is not null, and has a length so it can't be empty or missing. Then if the value of the string 'ref' is somehow invalid (e.g. a value that's not in the list of branch name or tag name as that's the value it represents). Let's say ref = 'invalidRef', the current tests cases with requester.send() would return the json file - _repos-hub4j-test-org_ghworkflowtest_actions_workflows_6817859_dispatches-5.json_ and since it has the status code 422, the current test cases will fail at the last case with singletonMap as inputs.
Should I change to status code to 204, same as _repos-hub4j-test-org_ghworkflowtest_actions_workflows_6817859_dispatches-4.json_, which doesn't have the inputs, to fix the test? And for the invalid ref, what would you say would be best way to fix it?

Reply:
@yixingz3 
Please try testing manually and recording _new_ data. See CONTRIBUTING.md for how to do this.